### ISSUE TITLE
Installing containers without using the fabric deploy task doesnt set salt grains

### ISSUE THREAD

**Issue by niallcreech** [ministryofjustice/bootstrap-salt#59]
When we first run a stack creation and update containers are installed with tag 'latest' or the default_version pillar variable we supply. This means that the grain <container_name>_tag is not set, this is only done when we run the fabric 'deploy' task. As moj-docker-deploy relies on this tag for its docker.pulled state on the images and service container restarts, we can end up pulling images versions we're not going to use, and restarting containers that havent changes. 

To fix we could enforce the grain <container_name>_tag as the one true source of which container versions are installed, and it should always be set if we've installed any docker image through any pathway.

    asg = get_connection(Autoscale)
    asg.set_autoscaling_group(get_stack_name())
    asg.set_tag('{0}_tag'.format(container_name), tag)
    asg.set_tag('{0}_task'.format(container_name), container_task)

**Comment by bennythejudge** [comment]
After discussing this issue with @niallcreech, studying the code and running some tests, these are the conclusions:
1) the problem could not be replicated: I have:
- created a new stack with one container
- immediately after I have modified the pillar adding a second application and highstated - the first container was not restarted
- made some changes to the configuration of the first container (ports and env vars), and the container was (as expected) restarted
- repeated the highstate without any change - the containers have not been restarted.

2) there is no "clean" way to add tags to ASG during the stages of the highstate.

This issue could be closed now and reopened in the future if the problem arises again.

**Comment by niallcreech** [comment]
@bennythejudge has tested the scenarios that could cause an issue and when looking at the detail of actually implementing ASG tagging in our initial creation workflow (excluding a deploy) it turned out to be,

* a significant amount of work
* not very easy to do cleanly, we don't have easy access to salt pillar data during stack creation (pre-first highstate), and we'd need to do some extra work to to make the first highstate run (which pulls in our initial containers) set ASG tags , whereas other runs with tags already set did not override any ASG tags. 

In short, although possible to ensure ASG tagging always matches deployed containers at all points of stack existence. In practice it would require more work than warranted for this feature.

**Comment by KrisSaxton** [comment]
Closing: unable to replicate.
